### ISSUE TITLE
False negative in `if-expression`

### ISSUE THREAD

**Issue by theRealProHacker** [dart-lang/linter#3315]
## Describe the issue
Dart doesn't lint, the last statement in the `innerFunc` below. It only gives some `infos` because I am not using the variables anywhere. However, building the project fails dramatically.

## To Reproduce
```dart
void main() {
  void innerFunc(bool myBool, String errorMessage){
    String? _hint = !myBool ? "Value1" : null;
    String? _label = myBool ? "Value2" : null;
    String? _errorText = errorMessage.isNotEmpty ? errorMessage;
  }
}
```
## Expected behavior
I expect the linter to give me a clear warning, at that line, and tell me exactly why the error is happening. Even building gives you a quite unclear parse error: `Error: This couldn't be parsed.` Which doesn't really tell my anything about the error. Instead, the linter or at least the compiler should tell me that I am missing the `:` which most other modern linters/compiler/interpreters are able to do and which is pretty easy because after an `?` the compiler should be expecting an `:`.
